pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2167,3 +2167,35 @@ def test_merge_datetime_upcast_dtype():
         }
     )
     tm.assert_frame_equal(result, expected)
+
```

Comment:
can you locate this near any other tests for right merge (or may not be any, if so pls comment)